dali/pipeline/operators/reader/loader/loader.h

```diff
@@ -71,6 +72,14 @@ class Loader {
 
   virtual ~Loader() = default;
 
+  // We need this two stage init because overriden PrepareMetadata
+  // is not known in Loader ctor
+  void Init() {
+    if (!lazy_) {
+      PrepareMetadata();
```

Comment:
Maybe wrap this with `std::cal_once` also?

Reply:
Agreed ;)